CNFE when using Smallrye @ConfigMapping

**Describe the bug**
I am using configmappings for a Map<String, Configuration>.
https://smallrye.io/docs/smallrye-config/mapping/mapping.html#_collections_config_mapping

When I run in dev mode, I get a CNFE for my `Configuration` class.
When I run as jar (mvn clean install), everything works as expected.

**Expected behavior**
ConfigMapping works in dev mode as it does as jar.

**Actual behavior**
```
2020-10-16 14:03:07,817 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile dev): java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:675)
        at io.quarkus.runtime.Application.start(Application.java:90)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:93)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:62)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:104)
        at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.runner.bootstrap.StartupActionImpl$3.run(StartupActionImpl.java:134)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.NoClassDefFoundError: org/acme/amqp/config/ConnectionConfiguration
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.System$2.defineClass(System.java:2128)
        at java.base/java.lang.invoke.MethodHandles$Lookup.defineClass(MethodHandles.java:962)
        at io.smallrye.common.classloader.ClassDefiner$1.run(ClassDefiner.java:19)
        at io.smallrye.common.classloader.ClassDefiner$1.run(ClassDefiner.java:14)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at io.smallrye.common.classloader.ClassDefiner.defineClass(ClassDefiner.java:14)
        at io.smallrye.config.ConfigMappingLoader.loadClass(ConfigMappingLoader.java:76)
        at io.smallrye.config.ConfigMappingLoader.loadClass(ConfigMappingLoader.java:71)
        at io.smallrye.config.ConfigMappingLoader.getImplementationClass(ConfigMappingLoader.java:61)
        at io.smallrye.config.ConfigMappingLoader$1.computeValue(ConfigMappingLoader.java:18)
        at io.smallrye.config.ConfigMappingLoader$1.computeValue(ConfigMappingLoader.java:15)
        at java.base/java.lang.ClassValue.getFromHashMap(ClassValue.java:226)
        at java.base/java.lang.ClassValue.getFromBackup(ClassValue.java:208)
        at java.base/java.lang.ClassValue.get(ClassValue.java:114)
        at io.smallrye.config.ConfigMappingLoader.configMappingObject(ConfigMappingLoader.java:37)
        at io.smallrye.config.ConfigMappingContext.constructGroup(ConfigMappingContext.java:68)
        at io.smallrye.config.ConfigMappingProvider$GetOrCreateEnclosingGroupInMap.apply(ConfigMappingProvider.java:507)
        at io.smallrye.config.ConfigMappingProvider$GetOrCreateEnclosingGroupInMap.accept(ConfigMappingProvider.java:513)
        at io.smallrye.config.ConfigMappingProvider$GetOrCreateEnclosingGroupInMap.accept(ConfigMappingProvider.java:477)
        at io.smallrye.config.ConfigMappingProvider$ConsumeOneAndThen.accept(ConfigMappingProvider.java:175)
        at io.smallrye.config.ConfigMappingProvider$ConsumeOneAndThen.accept(ConfigMappingProvider.java:166)
        at io.smallrye.config.ConfigMappingProvider.mapConfiguration(ConfigMappingProvider.java:633)
        at io.smallrye.config.ConfigMappingProvider.mapConfiguration(ConfigMappingProvider.java:602)
        at io.smallrye.config.ConfigMappings.registerConfigMappings(ConfigMappings.java:44)
        at io.smallrye.config.ConfigMappings.registerConfigMappings(ConfigMappings.java:32)
        at io.quarkus.arc.runtime.ConfigRecorder.registerConfigMappings(ConfigRecorder.java:52)
        at io.quarkus.deployment.steps.ConfigBuildStep$registerConfigMappings510524965.deploy_0(ConfigBuildStep$registerConfigMappings510524965.zig:130)
        at io.quarkus.deployment.steps.ConfigBuildStep$registerConfigMappings510524965.deploy(ConfigBuildStep$registerConfigMappings510524965.zig:40)
        at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:532)
        ... 12 more
Caused by: java.lang.ClassNotFoundException: org.acme.amqp.config.ConnectionConfiguration
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:412)
        at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:365)
        ... 42 more
```


**To Reproduce**

Steps to reproduce the behavior:
1. Download the reproducer:
[configmapping-cnfe.zip](https://github.com/quarkusio/quarkus/files/5391388/configmapping-cnfe.zip)
2. mvn quarkus:dev
3. CNFE from above happens
4. Now, run as jar (mvn quarkus:dev cd target/ java -jar  configmapping-cnfe-1.0.0-SNAPSHOT-runner.jar)
5. Quarkus starts.
6. Go to http://localhost:8080/hello and you see the configuration value.

**Configuration**
```properties
rabbitmq.wolf.connection-string=https://example.com

```


**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
MSYS_NT-10.0 NANB7NLNVP2 2.10.0(0.325/5/3) 2018-06-13 23:34 x86_64 Msys
 - Quarkus version or git rev: 1.9.0.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: C:\eclipse\tools\apache-maven\bin\..
Java version: 11.0.7, vendor: Azul Systems, Inc., runtime: C:\eclipse\tools\zulu11.39.15-ca-jdk11.0.7-win_x64
Default locale: de_DE, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -34,7 +34,7 @@
         <microprofile-reactive-streams-operators.version>1.0.1</microprofile-reactive-streams-operators.version>
         <microprofile-rest-client.version>1.4.1</microprofile-rest-client.version>
         <smallrye-common.version>1.4.0</smallrye-common.version>
-        <smallrye-config.version>1.9.1</smallrye-config.version>
+        <smallrye-config.version>1.9.2</smallrye-config.version>
         <smallrye-health.version>2.2.5</smallrye-health.version>
         <smallrye-metrics.version>2.4.4</smallrye-metrics.version>
         <smallrye-open-api.version>2.0.12</smallrye-open-api.version>

--- a/core/runtime/src/main/java/io/quarkus/runtime/configuration/Substitutions.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/configuration/Substitutions.java
@@ -57,7 +57,7 @@ final class Substitutions {
     @TargetClass(className = "io.smallrye.config.ConfigMappingLoader")
     static final class Target_ConfigMappingLoader {
         @Substitute
-        static Class<?> loadClass(final String className, final byte[] classBytes) {
+        private static Class<?> defineClass(final Class<?> parent, final String className, final byte[] classBytes) {
             return null;
         }
     }

